### PR TITLE
API FB (via WebScraping) & tests

### PR DESCRIPTION
**Pull request by RomTamam**

### CODE REVIEW

**Comment by simzacks** [comment]
Have you tried executing the methods in the file API_FB.py class Manipulation_FB?
have you tried calling the main function in test_API_FB.py?
Have you executed the tests in tests.py?

**Comment by RomTamam** [comment]
Have you tried executing the methods in the file API_FB.py class Manipulation_FB? **Yes and it worked**
have you tried calling the main function in test_API_FB.py? **Yes and it worked**
Have you executed the tests in tests.py? **didn't succeed this one yet... still checking it**

btw, can I get an explanation about your last comment regarding the tests:
"That list of outputs is very temporary as jobs change all the time.
For testing purposes, it might make sense to copy their HTML page statically so you can mock the tests."

thanks!

**Comment by simzacks** [comment]
I'm beginning to understand what you did here. 
Basically, while your functions are defined within a class, you're really not using class constructs.
My suggestion would be:
```
from bs4 import BeautifulSoup
import requests


class Manipulation_FB:
    def __init__(self):
        self.soup = None
        self.request = None

    def extract_jobs_url(self):
        fb_url = "https://www.facebook.com"
        jobs_links = []
        list_jobs_links = []
        jobs_list = self.soup.find("div", {"class": "_8tk7"})
        jobs = jobs_list.find_all("a", {"class": "_8sef"})
        links = [link['href'] for link in jobs]
        jobs_links = links
        for link in jobs_links:
            list_jobs_links.append(fb_url+link)
            # Need to push to a db
        print(list_jobs_links)
        return list_jobs_links

    def extract_jobs_title(self):
        jobs_titles = []
        list_jobs_titles = []
        jobs_list = self.soup.find("div", {"class": "_8tk7"})
        jobs = jobs_list.find_all("div", {"class": "_8sel"})
        jobs_titles = jobs
        for title in jobs_titles:
            list_jobs_titles.append(title.get_text())
            # Need to push to a db
        print(list_jobs_titles)
        return list_jobs_titles

    def create_soup(self):
        print(self.request.status_code)
        soup = BeautifulSoup(self.request.content, 'html.parser')
        # print(soup.prettify())
        self.soup = soup

    def create_req():
        # Load our first page
        # Need to pull from db
        search_url = "https://www.facebook.com/careers/jobs/?q="
        prefrences = "DevOps"
        r = requests.get(search_url+prefrences)
        self.request =  r
```

**Comment by simzacks** [comment]
To use this class you would then do:
manipulation = Manipulation_FB()
manipulation.create_req()
manipulation.create_soup()
manipulation.extract_jobs_url()
manipulation.extract_jobs_title()

File: Pipfile [hunk 1]
```diff
@@ -10,6 +10,8 @@ pytest = "*"
 [packages]
 django = "*"
```
Comment (Yarboa) [inline Pipfile:11]: Tried to run your code and i failed on /bin/bash setup.sh
Please update you Pipfile.lock
remove/ copy Pipfile.lock
Run:
* pipenv clean
* pipenv lock

File: pytest.ini [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[pytest]
+DJANGO_SETTINGS_MODULE = ultimatejobweb.settings
```
Comment (Yarboa) [inline pytest.ini:2]: Setting is located under ultimatejob.setting, did you try running the tests?

File: test_API_FB.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# Import and load necessary lib
+from API_FB import Manipulation_FB
```
Comment (Yarboa) [inline test_API_FB.py:2]: Why do you need this file, you should have tests to check it

Comment (Yarboa) [reply test_API_FB.py:2]: test_API_FB.py it should not be in project

File: pytest.ini (hunk 2 above)
Comment (Yarboa) [reply pytest.ini:2]: If you use this settings, it is for pipenv install pipenv install --dev pytest-django and not pytest, compare your Pipfile

**Review by Yarboa — COMMENTED** [review]
While i was running pipenv run pytest -v
I got the following error, 
/home/vagrant/.local/share/virtualenvs/vagrant-gKDsaKU3/lib64/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests.py:2: in <module>
    from ultimatejobweb.API_FB import Manipulation_FB
E   ModuleNotFoundError: No module named 'ultimatejobweb.API_FB'

Did you try it?

**Pull request closed without merging**